src/grep.rs

```diff
@@ -89,7 +89,7 @@ fn grep_bytes(
       write!(stdout, "{}: ", path)?;
     }
 
-    stdout.write(bs)?;
+    stdout.write_all(bs)?;
```

Comment:
I believe this solves a bug. A call to `write` that doesn't output all bytes would be considered successful. Many thanks for finding this.